File: ColegioTrenerAndroid/app/src/main/java/com/jjmf/colegiotrenerandroid/data/dto/EstadoAutorizacionDto.kt
package com.jjmf.colegiotrenerandroid.data.dto

import com.google.gson.annotations.SerializedName
import com.jjmf.colegiotrenerandroid.domain.model.EstadoAutorizacion

data class EstadoAutorizacionDto(
    @SerializedName("anoaca") val anoaca: String?,
    @SerializedName("ctacli") val ctacli: String?,
    @SerializedName("nombre") val nombre: String?,
    @SerializedName("ctamae") val ctamae: String?,
    @SerializedName("codgra") val codgra: String?,
    @SerializedName("autorizo") val autorizo: Double?,
) {
    fun toDomain(): EstadoAutorizacion {
        return EstadoAutorizacion(
            anoaca = anoaca?.trim(),
            ctacli = ctacli?.trim(),
            nombre = nombre?.trim(),
            ctamae = ctamae?.trim(),
            codgra = codgra?.trim(),
            autorizo = autorizo == 0.0
        )
    }
}

File: ColegioTrenerSwift/ColegioTrenerSwift/Network/Dto/EstadoAutorizacionDto.swift
//
//  EstadoAutorizacionDto.swift
//  ColegioTrenerSwift
//
//  Created by Joel Maldonado on 1/05/24.
//

import SwiftUI

struct EstadoAutorizacionDto: Codable {
    let anoaca: String?
    let ctacli: String?
    let nombre: String?
    let ctamae: String?
    let codgra: String?
    let autorizo: Double?
    
    func toDomain() -> EstadoAutorizacion {
        return EstadoAutorizacion(
            anoaca: anoaca?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            ctacli: ctacli?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            nombre: nombre?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            ctamae: ctamae?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            codgra: codgra?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            autorizo: autorizo == 1.0
        )
    }
}
